src/app/core/market/api/cart/cart.service.ts

```diff
@@ -15,13 +16,17 @@ export class CartService {
 
   constructor(
     private market: MarketService,
+    private marketState: MarketStateService,
     private snackbar: SnackbarService
   ) { }
 
   addItem(listingItemId: number) {
     this.log.d(`Adding listingItemId=${listingItemId} to cart with id=1`);
-    this.market.call('cartitem', ['add', 1, listingItemId]).take(1).do(
-      data => this.snackbar.open(data),
+    this.market.call('cartitem', ['add', 1, listingItemId]).take(1).subscribe(
+      data => {
+        this.snackbar.open(data)
+        this.marketState.registerStateCall('cart', null, ['get', 1])
```

Comment:
Can we move this into the cart.service?

Reply:
Should update: on addition, on delete, on clear

Reply:
lol it seems already in cart service :P